lib/board/CMakeLists.txt

```diff
@@ -22,6 +22,14 @@ set(sources
     util.c
     variant.c)
 
+if(NOT ${KK_HAVE_STRLCPY})
```

Comment:
```suggestion
if(NOT DEFINED ${KK_HAVE_STRLCPY})
```

Reply:
I don't think that's correct. CheckSymbolExists sets the variable to either `1` or `""` depending on whether it was found in the supplied header:

https://github.com/Kitware/CMake/blob/master/Modules/CheckSymbolExists.cmake#L124

And `if()` properly detects those two resulting values:

https://cmake.org/cmake/help/latest/command/if.html#condition-syntax